CesiumGltf/test/TestAccessorView.cpp

```diff
@@ -37,3 +37,37 @@ TEST_CASE("AccessorView construct and read example") {
 
   CHECK(firstPosition == glm::vec3(1.0f, 2.0f, 3.0f));
 }
+
+TEST_CASE("Create AccessorView of unknown type with lambda") {
+  using namespace CesiumGltf;
+
+  Model model;
+
+  Buffer& buffer = model.buffers.emplace_back();
+  buffer.cesium.data.resize(4);
+  buffer.cesium.data[0] = std::byte(1);
+  buffer.cesium.data[1] = std::byte(2);
+  buffer.cesium.data[2] = std::byte(3);
+  buffer.cesium.data[3] = std::byte(4);
+  buffer.byteLength = buffer.cesium.data.size();
+
+  BufferView& bufferView = model.bufferViews.emplace_back();
+  bufferView.buffer = 0;
+  bufferView.byteLength = buffer.byteLength;
+
+  Accessor& accessor = model.accessors.emplace_back();
+  accessor.bufferView = 0;
+  accessor.count = 1;
+
+  accessor.componentType = Accessor::ComponentType::UNSIGNED_INT;
+  createAccessorView(model, accessor, [](const auto& accessorView) {
```

Comment:
something weird happens in this test for `gcc` and `clang` at least. The auto is deduced to AccessorView<AccessorType::SCALAR<float>> instead of AccessorView<AccessorType::SCALAR<unsigned int>>. My initial guess is the deduction coming from std::invoke_result_t<> in the function signature